File: Leetcode/Leetcode-DSA-course/Hashing/2428-equal-row-and-column-pairs/equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        dic = defaultdict(int)
        for row in grid:
            dic[tuple(row)] += 1
        dic2 = defaultdict(int)
        for col in range(len(grid[0])):
            currentCol = []
            for row in range(len(grid)):
                currentCol.append(grid[row][col])
            dic2[tuple(currentCol)] += 1
        ans = 0
        for arr in dic:
            ans += dic[arr] * dic2[arr]
        return ans

File: Leetcode/Leetcode-DSA-course/Hashing/2428-equal-row-and-column-pairs/equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        map<vector<int>, int> dic;
        for (auto row : grid)
            dic[row]++;
        map<vector<int>, int> dic2;
        for (int col = 0; col < grid[0].size(); col++) {
            vector<int> currentCol;
            for (int row = 0; row < grid.size(); row++)
                currentCol.push_back(grid[row][col]);
            dic2[currentCol]++;
        }
        int ans = 0;
        for (auto [arr, val] : dic) {
            ans += val * dic2[arr];
        }
        return ans;
    }
};